pandas/tests/io/formats/test_format.py

```diff
@@ -3219,6 +3231,67 @@ def test_period_tz(self):
             per = dt.to_period(freq="H")
         assert per.format()[0] == "2013-01-01 00:00"
 
+    @pytest.mark.parametrize(
+        "locale_str",
+        [
+            pytest.param(None, id=str(locale.getlocale())),
+            "it_IT.utf8",
+            "it_IT",  # Note: encoding will be 'ISO8859-1'
+            "zh_CN.utf8",
+            "zh_CN",  # Note: encoding will be 'gb2312'
+        ],
+    )
+    def test_period_non_ascii_fmt(self, locale_str):
+        # GH#46468 non-ascii char in input format string leads to wrong output
+
+        # Skip if locale cannot be set
+        if locale_str is not None and not tm.can_set_locale(locale_str, locale.LC_ALL):
+            pytest.skip(f"Skipping as locale '{locale_str}' cannot be set on host.")
+
+        # Change locale temporarily for this test.
+        with tm.set_locale(locale_str, locale.LC_ALL) if locale_str else nullcontext():
```

Comment:
```suggestion
        with tm.set_locale(locale_str, locale.LC_ALL):
```

The condition above should ensure we can set the locale?

Reply:
No, this code is to handle the case when the param `locale_str` is set to `None`. This is the first parametrization value that is tried (see pytest parametrize decorator. It means "do not set locale for this test".

Reply:
Ah okay thanks for the explaination